docs/sql-programming-guide.md

```diff
@@ -421,9 +421,10 @@ In the simplest form, the default data source (`parquet` unless otherwise config
 
 You can also manually specify the data source that will be used along with any extra options
 that you would like to pass to the data source. Data sources are specified by their fully qualified
-name (i.e., `org.apache.spark.sql.parquet`), but for built-in sources you can also use their short
-names (`json`, `parquet`, `jdbc`, `orc`, `libsvm`, `csv`, `text`). DataFrames loaded from any data
-source type can be converted into other types using this syntax.
+name (i.e., `org.apache.spark.sql.parquet`) or short name specified in
+`DataSourceRegister.shortName()` in their implementation. For built-in sources you can use their
```

Comment:
hm this isn't clear to the end user at all, since there is no class called DataSourceRegister even in the source code for data sources. I'd just leave it out. It should be the job of the data source to tell users how to use it.